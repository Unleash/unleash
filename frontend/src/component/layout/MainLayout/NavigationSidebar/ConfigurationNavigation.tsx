import type React from 'react';
import type { FC } from 'react';
import type { NavigationMode } from './NavigationMode.ts';
import { Typography } from '@mui/material';
import Accordion from '@mui/material/Accordion';
import AccordionDetails from '@mui/material/AccordionDetails';
import AccordionSummary from '@mui/material/AccordionSummary';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';

const AccordionHeader: FC<{ children?: React.ReactNode }> = ({ children }) => {
    return (
        <AccordionSummary
            expandIcon={<ExpandMoreIcon />}
            aria-controls='configure-content'
            id='configure-header'
        >
            <Typography sx={{ fontWeight: 'bold', fontSize: 'small' }}>
                {children}
            </Typography>
        </AccordionSummary>
    );
};

export const ConfigurationNavigation: FC<{
    expanded: boolean;
    onExpandChange: (expanded: boolean) => void;
    mode: NavigationMode;
    title: string;
    children?: React.ReactNode;
}> = ({ mode, expanded, onExpandChange, title, children }) => {
    return (
        <Accordion
            disableGutters={true}
            sx={{
                boxShadow: 'none',
                '&:before': {
                    display: 'none',
                },
            }}
            expanded={expanded}
            onChange={(_, expand) => {
                onExpandChange(expand);
            }}
        >
            {mode === 'full' && <AccordionHeader>{title}</AccordionHeader>}
            <AccordionDetails sx={{ p: 0 }}>{children}</AccordionDetails>
        </Accordion>
    );
};
