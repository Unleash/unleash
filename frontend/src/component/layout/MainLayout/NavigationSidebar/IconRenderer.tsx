import EmptyIcon from '@mui/icons-material/CheckBoxOutlineBlankOutlined';
import type SvgIcon from '@mui/material/SvgIcon/SvgIcon';
import ApplicationsIcon from '@mui/icons-material/AppsOutlined';
import ContextFieldsIcon from '@mui/icons-material/AccountTreeOutlined';
import FlagTypesIcon from '@mui/icons-material/OutlinedFlag';
import IntegrationsIcon from '@mui/icons-material/IntegrationInstructionsOutlined';
import SegmentsIcon from '@mui/icons-material/DonutLargeOutlined';
import CustomStrategiesIcon from '@mui/icons-material/ExtensionOutlined';
import TagTypesIcon from '@mui/icons-material/LabelImportantOutlined';
import EnvironmentsIcon from '@mui/icons-material/CloudOutlined';
import UsersIcon from '@mui/icons-material/GroupOutlined';
import ServiceAccountIcon from '@mui/icons-material/Computer';
import GroupsIcon from '@mui/icons-material/GroupsOutlined';
import RoleIcon from '@mui/icons-material/AdminPanelSettingsOutlined';
import SearchIcon from '@mui/icons-material/Search';
import ApiAccessIcon from '@mui/icons-material/KeyOutlined';
import SingleSignOnIcon from '@mui/icons-material/AssignmentOutlined';
import NetworkIcon from '@mui/icons-material/HubOutlined';
import MaintenanceIcon from '@mui/icons-material/BuildOutlined';
import BannersIcon from '@mui/icons-material/ViewCarousel';
import InstanceStatsIcon from '@mui/icons-material/QueryStatsOutlined';
import LicenseIcon from '@mui/icons-material/ReceiptLongOutlined';
import InstancePrivacyIcon from '@mui/icons-material/ShieldOutlined';
import LoginHistoryIcon from '@mui/icons-material/HistoryOutlined';
import CorsIcon from '@mui/icons-material/StorageOutlined';
import BillingIcon from '@mui/icons-material/CreditCardOutlined';
import EventLogIcon from '@mui/icons-material/EventNoteOutlined';
import GitHubIcon from '@mui/icons-material/GitHub';
import LibraryBooksIcon from '@mui/icons-material/LibraryBooks';
import type { FC } from 'react';

// TODO: move to routes
const icons: Record<string, typeof SvgIcon> = {
    '/search': SearchIcon,
    '/applications': ApplicationsIcon,
    '/context': ContextFieldsIcon,
    '/feature-toggle-type': FlagTypesIcon,
    '/integrations': IntegrationsIcon,
    '/segments': SegmentsIcon,
    '/strategies': CustomStrategiesIcon,
    '/tag-types': TagTypesIcon,
    '/environments': EnvironmentsIcon,
    '/admin/users': UsersIcon,
    '/admin/service-accounts': ServiceAccountIcon,
    '/admin/groups': GroupsIcon,
    '/admin/roles': RoleIcon,
    '/admin/api': ApiAccessIcon,
    '/admin/auth': SingleSignOnIcon,
    '/admin/network': NetworkIcon,
    '/admin/maintenance': MaintenanceIcon,
    '/admin/banners': BannersIcon,
    '/admin/instance': InstanceStatsIcon,
    '/admin/license': LicenseIcon,
    '/admin/instance-privacy': InstancePrivacyIcon,
    '/admin/logins': LoginHistoryIcon,
    '/admin/cors': CorsIcon,
    '/admin/billing': BillingIcon,
    '/history': EventLogIcon,
    GitHub: GitHubIcon,
    Documentation: LibraryBooksIcon,
};

const findIcon = (key: string) => {
    return icons[key] || EmptyIcon;
};

export const IconRenderer: FC<{ path: string }> = ({ path }) => {
    const IconComponent = findIcon(path); // Fallback to 'default' if the type is not found

    return <IconComponent />;
};
