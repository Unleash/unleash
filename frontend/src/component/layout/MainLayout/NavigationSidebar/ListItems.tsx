import type React from 'react';
import type { FC, ReactNode } from 'react';
import {
    ListItem,
    ListItemButton,
    ListItemIcon,
    ListItemText,
    styled,
    Tooltip,
    Typography,
} from '@mui/material';
import { Link } from 'react-router-dom';
import { basePath } from 'utils/formatPath';
import SignOutIcon from '@mui/icons-material/ExitToApp';
import type { Theme } from '@mui/material/styles/createTheme';

const listItemButtonStyle = (theme: Theme) => ({
    borderRadius: theme.spacing(0.5),
    borderLeft: `${theme.spacing(0.5)} solid transparent`,
    '&.Mui-selected': {
        borderLeft: `${theme.spacing(0.5)} solid ${theme.palette.primary.main}`,
    },
});

const CappedText = styled(Typography)({
    whiteSpace: 'nowrap',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    width: '100%',
});

const StyledListItemIcon = styled(ListItemIcon)(({ theme }) => ({
    minWidth: theme.spacing(4),
    margin: theme.spacing(0.25, 0),
}));

const StyledListItemText = styled(ListItemText)(({ theme }) => ({
    margin: 0,
}));

export const FullListItem: FC<{
    href: string;
    text: string;
    badge?: ReactNode;
    onClick: () => void;
    selected?: boolean;
    children?: React.ReactNode;
}> = ({ href, text, badge, onClick, selected, children }) => {
    return (
        <ListItem disablePadding onClick={onClick}>
            <ListItemButton
                dense={true}
                component={Link}
                to={href}
                sx={listItemButtonStyle}
                selected={selected}
            >
                <StyledListItemIcon>{children}</StyledListItemIcon>
                <StyledListItemText>
                    <CappedText>{text}</CappedText>
                </StyledListItemText>
                {badge}
            </ListItemButton>
        </ListItem>
    );
};

export const ExternalFullListItem: FC<{
    href: string;
    text: string;
    children?: React.ReactNode;
}> = ({ href, text, children }) => {
    return (
        <ListItem disablePadding>
            <ListItemButton
                dense={true}
                component={Link}
                to={href}
                sx={listItemButtonStyle}
                rel='noopener noreferrer'
                target='_blank'
            >
                <StyledListItemIcon>{children}</StyledListItemIcon>
                <StyledListItemText>
                    <CappedText>{text}</CappedText>
                </StyledListItemText>
            </ListItemButton>
        </ListItem>
    );
};
export const SignOutItem = () => {
    return (
        <form method='POST' action={`${basePath}/logout`}>
            <ListItem disablePadding>
                <ListItemButton
                    dense={true}
                    component='button'
                    type='submit'
                    sx={listItemButtonStyle}
                >
                    <StyledListItemIcon>
                        <SignOutIcon />
                    </StyledListItemIcon>
                    <StyledListItemText>
                        <CappedText>Sign out</CappedText>
                    </StyledListItemText>
                </ListItemButton>
            </ListItem>
        </form>
    );
};

export const MiniListItem: FC<{
    href: string;
    text: string;
    selected?: boolean;
    onClick: () => void;
    children?: React.ReactNode;
}> = ({ href, text, selected, onClick, children }) => {
    return (
        <ListItem disablePadding onClick={onClick}>
            <ListItemButton
                dense={true}
                component={Link}
                to={href}
                sx={listItemButtonStyle}
                selected={selected}
            >
                <Tooltip title={text} placement='right'>
                    <StyledListItemIcon>{children}</StyledListItemIcon>
                </Tooltip>
            </ListItemButton>
        </ListItem>
    );
};
