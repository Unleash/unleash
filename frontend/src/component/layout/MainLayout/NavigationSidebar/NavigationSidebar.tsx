import { Box, styled } from '@mui/material';
import { type FC, useState, useEffect } from 'react';
import { useNavigationMode } from './useNavigationMode';
import { ShowAdmin, ShowHide } from './ShowHide';
import { useRoutes } from './useRoutes';
import { useExpanded } from './useExpanded';
import {
    OtherLinksList,
    PrimaryNavigationList,
    RecentFlagsNavigation,
    RecentProjectsNavigation,
    SecondaryNavigation,
    SecondaryNavigationList,
} from './NavigationList';
import { useInitialPathname } from './useInitialPathname';
import { useLastViewedProject } from 'hooks/useLastViewedProject';
import { useLastViewedFlags } from 'hooks/useLastViewedFlags';
import { NewInUnleash } from './NewInUnleash/NewInUnleash';

export const MobileNavigationSidebar: FC<{ onClick: () => void }> = ({
    onClick,
}) => {
    const { routes } = useRoutes();

    return (
        <>
            <NewInUnleash onItemClick={onClick} />
            <PrimaryNavigationList mode='full' onClick={onClick} />
            <SecondaryNavigationList
                routes={routes.mainNavRoutes}
                mode='full'
                onClick={onClick}
            />
            <SecondaryNavigationList
                routes={routes.adminRoutes}
                mode='full'
                onClick={onClick}
            />
            <OtherLinksList />
        </>
    );
};

export const StretchContainer = styled(Box)<{ mode: string }>(
    ({ theme, mode }) => ({
        backgroundColor: theme.palette.background.paper,
        padding: theme.spacing(2),
        alignSelf: 'stretch',
        display: 'flex',
        flexDirection: 'column',
        gap: theme.spacing(2),
        zIndex: 1,
        overflowAnchor: 'none',
        minWidth: mode === 'full' ? theme.spacing(40) : 'auto',
        width: mode === 'full' ? theme.spacing(40) : 'auto',
    }),
);

// This component is needed when the sticky item could overlap with nav items. You can replicate it on a short screen.
const StickyContainer = styled(Box)(({ theme }) => ({
    position: 'sticky',
    paddingBottom: theme.spacing(1.5),
    paddingTop: theme.spacing(1),
    bottom: theme.spacing(0),
    backgroundColor: theme.palette.background.paper,
    borderTop: `1px solid ${theme.palette.divider}`,
}));

export const NavigationSidebar = () => {
    const { routes } = useRoutes();

    const [mode, setMode] = useNavigationMode();
    const [expanded, changeExpanded] = useExpanded<'configure' | 'admin'>();
    const initialPathname = useInitialPathname();

    const [activeItem, setActiveItem] = useState(initialPathname);

    const { lastViewed: lastViewedProject } = useLastViewedProject();
    const showRecentProject = mode === 'full' && lastViewedProject;

    const { lastViewed: lastViewedFlags } = useLastViewedFlags();
    const showRecentFlags = mode === 'full' && lastViewedFlags.length > 0;

    useEffect(() => {
        setActiveItem(initialPathname);
    }, [initialPathname]);

    return (
        <StretchContainer mode={mode}>
            <PrimaryNavigationList
                mode={mode}
                onClick={setActiveItem}
                activeItem={activeItem}
            />
            <SecondaryNavigation
                expanded={expanded.includes('configure')}
                onExpandChange={(expand) => {
                    changeExpanded('configure', expand);
                }}
                mode={mode}
                title='Configure'
            >
                <SecondaryNavigationList
                    routes={routes.mainNavRoutes}
                    mode={mode}
                    onClick={setActiveItem}
                    activeItem={activeItem}
                />
            </SecondaryNavigation>
            {mode === 'full' && (
                <SecondaryNavigation
                    expanded={expanded.includes('admin')}
                    onExpandChange={(expand) => {
                        changeExpanded('admin', expand);
                    }}
                    mode={mode}
                    title='Admin'
                >
                    <SecondaryNavigationList
                        routes={routes.adminRoutes}
                        mode={mode}
                        onClick={setActiveItem}
                        activeItem={activeItem}
                    />
                </SecondaryNavigation>
            )}

            {mode === 'mini' && (
                <ShowAdmin
                    onChange={() => {
                        changeExpanded('admin', true);
                        setMode('full');
                    }}
                />
            )}

            {showRecentProject && (
                <RecentProjectsNavigation
                    mode={mode}
                    projectId={lastViewedProject}
                    onClick={() => setActiveItem('/projects')}
                />
            )}

            {showRecentFlags && (
                <RecentFlagsNavigation
                    mode={mode}
                    flags={lastViewedFlags}
                    onClick={() => setActiveItem('/projects')}
                />
            )}

            {/* this will push the show/hide to the bottom on short nav list */}
            <Box sx={{ flex: 1 }} />

            <StickyContainer>
                <NewInUnleash
                    mode={mode}
                    onMiniModeClick={() => setMode('full')}
                />
                <ShowHide
                    mode={mode}
                    onChange={() => {
                        setMode(mode === 'full' ? 'mini' : 'full');
                    }}
                />
            </StickyContainer>
        </StretchContainer>
    );
};
