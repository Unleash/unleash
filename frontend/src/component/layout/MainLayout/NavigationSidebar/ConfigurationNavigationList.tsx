import type { FC } from 'react';
import type { INavigationMenuItem } from 'interfaces/route';
import type { NavigationMode } from './NavigationMode.ts';
import { MenuListItem } from './ListItems.tsx';
import { List } from '@mui/material';
import { IconRenderer } from './IconRenderer.tsx';
import { useUiFlag } from 'hooks/useUiFlag.ts';
import StopRoundedIcon from '@mui/icons-material/StopRounded';
import { useShowBadge } from 'component/layout/components/EnterprisePlanBadge/useShowBadge';
import { EnterprisePlanBadge } from 'component/layout/components/EnterprisePlanBadge/EnterprisePlanBadge';

export const ConfigurationNavigationList: FC<{
    routes: INavigationMenuItem[];
    mode: NavigationMode;
    onClick: (activeItem: string) => void;
    activeItem?: string;
}> = ({ routes, mode, onClick, activeItem }) => {
    const showBadge = useShowBadge();
    const sideMenuCleanup = useUiFlag('sideMenuCleanup');

    return (
        <List>
            {routes.map((route) => (
                <MenuListItem
                    key={route.title}
                    onClick={() => onClick(route.path)}
                    href={route.path}
                    text={route.title}
                    selected={activeItem === route.path}
                    badge={
                        showBadge(route?.menu?.mode) ? (
                            <EnterprisePlanBadge />
                        ) : null
                    }
                    mode={mode}
                    icon={
                        sideMenuCleanup ? (
                            <StopRoundedIcon fontSize='small' color='primary' />
                        ) : (
                            <IconRenderer path={route.path} />
                        )
                    }
                    secondary={sideMenuCleanup}
                />
            ))}
        </List>
    );
};
