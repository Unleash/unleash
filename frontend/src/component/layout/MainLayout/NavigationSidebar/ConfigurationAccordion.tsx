import { type FC, useEffect, useState } from 'react';
import { MenuListAccordion } from './ListItems.tsx';
import { useExpanded } from './useExpanded.ts';
import type { NavigationMode } from './NavigationMode.tsx';
import { IconRenderer } from './IconRenderer.tsx';
import { ConfigurationNavigationList } from './ConfigurationNavigationList.tsx';
import { useRoutes } from './useRoutes.ts';

type ConfigurationAccordionProps = {
    mode: NavigationMode;
    setMode;
    activeItem?: string;
    onClick?: () => void;
};

export const ConfigurationAccordion: FC<ConfigurationAccordionProps> = ({
    mode,
    setMode,
    activeItem,
    onClick,
}) => {
    const [expanded, changeExpanded] = useExpanded<'configure'>();
    const [temporarilyExpanded, setTemporarilyExpanded] = useState(false);
    const { routes } = useRoutes();
    const subRoutes = routes.mainNavRoutes;
    const hasActiveItem = Boolean(
        activeItem && subRoutes.some((route) => route.path === activeItem),
    );

    useEffect(() => {
        if (mode === 'mini') {
            setTemporarilyExpanded(false);
        }
    }, [mode]);

    const onExpandChange = () => {
        changeExpanded('configure', !expanded.includes('configure'));

        if (temporarilyExpanded) {
            setTemporarilyExpanded(false);
            setMode('mini');
        }
        if (mode === 'mini') {
            setTemporarilyExpanded(true);
            setMode('full');
        }
    };

    const onItemClick = () => {
        if (temporarilyExpanded) {
            setTemporarilyExpanded(false);
            setMode('mini');
        }
        onClick?.();
    };

    return (
        <MenuListAccordion
            title='Configure'
            expanded={
                (expanded.includes('configure') || temporarilyExpanded) &&
                mode === 'full'
            }
            onExpandChange={onExpandChange}
            mode={mode}
            icon={<IconRenderer path='Configure' />}
            active={hasActiveItem}
        >
            <ConfigurationNavigationList
                routes={subRoutes}
                mode={mode}
                onClick={onItemClick}
                activeItem={activeItem}
            />
        </MenuListAccordion>
    );
};
