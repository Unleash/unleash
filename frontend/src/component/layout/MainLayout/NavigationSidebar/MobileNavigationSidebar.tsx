import type { FC } from 'react';
import {
    PrimaryNavigationList,
    AdminSettingsLink,
    OtherLinksList,
} from './NavigationList.tsx';
import type { NewInUnleash } from './NewInUnleash/NewInUnleash.tsx';
import { ConfigurationNavigationList } from './ConfigurationNavigationList.tsx';
import { useRoutes } from './useRoutes.ts';
import { useUiFlag } from 'hooks/useUiFlag.ts';

export const MobileNavigationSidebar: FC<{
    onClick: () => void;
    NewInUnleash?: typeof NewInUnleash;
}> = ({ onClick, NewInUnleash }) => {
    const { routes } = useRoutes();
    const sideMenuCleanup = useUiFlag('sideMenuCleanup');

    return (
        <>
            {NewInUnleash ? <NewInUnleash /> : null}
            <PrimaryNavigationList
                mode='full'
                onClick={onClick}
                setMode={() => {}}
            />
            {!sideMenuCleanup ? (
                <ConfigurationNavigationList
                    routes={routes.mainNavRoutes}
                    mode='full'
                    onClick={onClick}
                />
            ) : null}
            <AdminSettingsLink mode={'full'} onClick={onClick} />
            <OtherLinksList />
        </>
    );
};
