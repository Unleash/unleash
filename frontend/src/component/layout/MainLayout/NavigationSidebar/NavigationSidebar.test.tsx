import { render } from 'utils/testRenderer';
import { NavigationSidebar } from './NavigationSidebar';
import { screen, fireEvent, waitFor } from '@testing-library/react';
import { createLocalStorage } from 'utils/createLocalStorage';
import { Route, Routes } from 'react-router-dom';
import { listItemButtonClasses as classes } from '@mui/material/ListItemButton';
import {
    type LastViewedFlag,
    useLastViewedFlags,
} from 'hooks/useLastViewedFlags';
import { type FC, useEffect } from 'react';
import { useLastViewedProject } from 'hooks/useLastViewedProject';
import { testServerRoute, testServerSetup } from 'utils/testServer';

const server = testServerSetup();

beforeEach(() => {
    window.localStorage.clear();
});

test('switch full mode and mini mode', () => {
    render(<NavigationSidebar />);

    expect(screen.queryByText('Projects')).toBeInTheDocument();
    expect(screen.queryByText('Applications')).toBeInTheDocument();
    expect(screen.queryByText('Users')).toBeInTheDocument();

    const hide = screen.getByText('Hide (⌘ + B)');

    fireEvent.click(hide);

    expect(screen.queryByText('Projects')).not.toBeInTheDocument();
    expect(screen.queryByText('Applications')).not.toBeInTheDocument();
    expect(screen.queryByText('Users')).not.toBeInTheDocument();

    const expand = screen.getByTestId('expand-navigation');
    fireEvent.click(expand);

    expect(screen.queryByText('Projects')).toBeInTheDocument();
    expect(screen.queryByText('Applications')).toBeInTheDocument();
    expect(screen.queryByText('Users')).toBeInTheDocument();
});

test('persist navigation mode and expansion selection in storage', async () => {
    render(<NavigationSidebar />);
    const { value } = createLocalStorage('navigation-mode:v1', {});
    expect(value).toBe('full');

    const configure = screen.getByText('Configure');
    configure.click(); // expand
    configure.click(); // hide
    const admin = screen.getByText('Admin');
    admin.click();

    const hide = screen.getByText('Hide (⌘ + B)');
    hide.click();

    await waitFor(() => {
        const { value } = createLocalStorage('navigation-mode:v1', {});
        expect(value).toBe('mini');

        const { value: expanded } = createLocalStorage(
            'navigation-expanded:v1',
            {},
        );
        expect(expanded).toEqual(['admin']);
    });
});

test('select active item', async () => {
    render(
        <Routes>
            <Route path={'/search'} element={<NavigationSidebar />} />
        </Routes>,
        { route: '/search' },
    );

    const links = screen.getAllByRole('link');

    expect(links[1]).toHaveClass(classes.selected);
});

const SetupComponent: FC<{ project: string; flags: LastViewedFlag[] }> = ({
    project,
    flags,
}) => {
    const { setLastViewed: setProject } = useLastViewedProject();
    const { setLastViewed: setFlag } = useLastViewedFlags();

    useEffect(() => {
        setProject(project);
        flags.forEach((flag) => {
            setFlag(flag);
        });
    }, []);

    return <NavigationSidebar />;
};

test('print recent projects and flags', async () => {
    testServerRoute(server, `/api/admin/projects/projectA/overview`, {
        name: 'projectNameA',
    });

    render(
        <SetupComponent
            project={'projectA'}
            flags={[{ featureId: 'featureA', projectId: 'projectB' }]}
        />,
    );

    await screen.findByText('projectNameA');
    await screen.findByText('featureA');
});
