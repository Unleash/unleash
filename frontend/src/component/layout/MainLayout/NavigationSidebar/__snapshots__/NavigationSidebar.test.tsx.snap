// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`order of items in navigation > menu for enterprise plan 1`] = `
[
  {
    "icon": "DashboardOutlinedIcon",
    "text": "Dashboard",
  },
  {
    "icon": "UnleashProjectIcon",
    "text": "Projects",
  },
  {
    "icon": "SearchIcon",
    "text": "Search",
  },
  {
    "icon": "AutoFixNormalIcon",
    "text": "Playground",
  },
  {
    "icon": "InsightsIcon",
    "text": "Insights",
  },
  {
    "icon": "AppsOutlinedIcon",
    "text": "Applications",
  },
  {
    "icon": "AccountTreeOutlinedIcon",
    "text": "Context fields",
  },
  {
    "icon": "OutlinedFlagIcon",
    "text": "Feature flag types",
  },
  {
    "icon": "ExtensionOutlinedIcon",
    "text": "Strategy types",
  },
  {
    "icon": "CloudOutlinedIcon",
    "text": "Environments",
  },
  {
    "icon": "LabelImportantOutlinedIcon",
    "text": "Tag types",
  },
  {
    "icon": "IntegrationInstructionsOutlinedIcon",
    "text": "Integrations",
  },
  {
    "icon": "DonutLargeOutlinedIcon",
    "text": "Segments",
  },
  {
    "icon": "GroupOutlinedIcon",
    "text": "Users",
  },
  {
    "icon": "ComputerIcon",
    "text": "Service accounts",
  },
  {
    "icon": "GroupsOutlinedIcon",
    "text": "Groups",
  },
  {
    "icon": "AdminPanelSettingsOutlinedIcon",
    "text": "Roles",
  },
  {
    "icon": "KeyOutlinedIcon",
    "text": "API access",
  },
  {
    "icon": "AssignmentOutlinedIcon",
    "text": "Single sign-on",
  },
  {
    "icon": "BuildOutlinedIcon",
    "text": "Maintenance",
  },
  {
    "icon": "ViewCarouselIcon",
    "text": "Banners",
  },
  {
    "icon": "QueryStatsOutlinedIcon",
    "text": "Instance stats",
  },
  {
    "icon": "ShieldOutlinedIcon",
    "text": "Instance privacy",
  },
  {
    "icon": "HistoryOutlinedIcon",
    "text": "Login history",
  },
  {
    "icon": "EventNoteOutlinedIcon",
    "text": "Event log",
  },
]
`;

exports[`order of items in navigation > menu for open-source 1`] = `
[
  {
    "icon": "DashboardOutlinedIcon",
    "text": "Dashboard",
  },
  {
    "icon": "UnleashProjectIcon",
    "text": "Projects",
  },
  {
    "icon": "SearchIcon",
    "text": "Search",
  },
  {
    "icon": "AutoFixNormalIcon",
    "text": "Playground",
  },
  {
    "icon": "AppsOutlinedIcon",
    "text": "Applications",
  },
  {
    "icon": "AccountTreeOutlinedIcon",
    "text": "Context fields",
  },
  {
    "icon": "OutlinedFlagIcon",
    "text": "Feature flag types",
  },
  {
    "icon": "ExtensionOutlinedIcon",
    "text": "Strategy types",
  },
  {
    "icon": "LabelImportantOutlinedIcon",
    "text": "Tag types",
  },
  {
    "icon": "IntegrationInstructionsOutlinedIcon",
    "text": "Integrations",
  },
  {
    "icon": "DonutLargeOutlinedIcon",
    "text": "Segments",
  },
  {
    "icon": "GroupOutlinedIcon",
    "text": "Users",
  },
  {
    "icon": "KeyOutlinedIcon",
    "text": "API access",
  },
  {
    "icon": "BuildOutlinedIcon",
    "text": "Maintenance",
  },
  {
    "icon": "QueryStatsOutlinedIcon",
    "text": "Instance stats",
  },
  {
    "icon": "ShieldOutlinedIcon",
    "text": "Instance privacy",
  },
  {
    "icon": "EventNoteOutlinedIcon",
    "text": "Event log",
  },
]
`;

exports[`order of items in navigation > menu for pro plan 1`] = `
[
  {
    "icon": "DashboardOutlinedIcon",
    "text": "Dashboard",
  },
  {
    "icon": "UnleashProjectIcon",
    "text": "Projects",
  },
  {
    "icon": "SearchIcon",
    "text": "Search",
  },
  {
    "icon": "AutoFixNormalIcon",
    "text": "Playground",
  },
  {
    "icon": "InsightsIcon",
    "text": "Insights",
  },
  {
    "icon": "AppsOutlinedIcon",
    "text": "Applications",
  },
  {
    "icon": "AccountTreeOutlinedIcon",
    "text": "Context fields",
  },
  {
    "icon": "OutlinedFlagIcon",
    "text": "Feature flag types",
  },
  {
    "icon": "ExtensionOutlinedIcon",
    "text": "Strategy types",
  },
  {
    "icon": "CloudOutlinedIcon",
    "text": "Environments",
  },
  {
    "icon": "LabelImportantOutlinedIcon",
    "text": "Tag types",
  },
  {
    "icon": "IntegrationInstructionsOutlinedIcon",
    "text": "Integrations",
  },
  {
    "icon": "DonutLargeOutlinedIcon",
    "text": "Segments",
  },
  {
    "icon": "GroupOutlinedIcon",
    "text": "Users",
  },
  {
    "icon": "ComputerIcon",
    "text": "Service accounts",
  },
  {
    "icon": "GroupsOutlinedIcon",
    "text": "Groups",
  },
  {
    "icon": "AdminPanelSettingsOutlinedIcon",
    "text": "Roles",
  },
  {
    "icon": "KeyOutlinedIcon",
    "text": "API access",
  },
  {
    "icon": "AssignmentOutlinedIcon",
    "text": "Single sign-on",
  },
  {
    "icon": "BuildOutlinedIcon",
    "text": "Maintenance",
  },
  {
    "icon": "ViewCarouselIcon",
    "text": "Banners",
  },
  {
    "icon": "QueryStatsOutlinedIcon",
    "text": "Instance stats",
  },
  {
    "icon": "ShieldOutlinedIcon",
    "text": "Instance privacy",
  },
  {
    "icon": "HistoryOutlinedIcon",
    "text": "Login history",
  },
  {
    "icon": "EventNoteOutlinedIcon",
    "text": "Event log",
  },
]
`;
